service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -240,6 +279,13 @@ private static List<TableMetadata> convertDatarepoTablesToCatalogTables(
         .toList();
   }
 
+  private static List<TableMetadata> convertRawlsTablesToCatalogTables(
+      Map<String, EntityTypeMetadata> entityTables) {
+    return entityTables.keySet().stream()
+        .map(entityTypeMetadata -> new TableMetadata().name(entityTypeMetadata).hasData(true))
```

Comment:
I don't think `hasData(true)` is correct. This should this be something like `hasData(entityTables.get(entityTypeMetadata).getCount() != 0)`.

Although if you do this, then this should use `entrySet().stream()` instead of `keySet().stream()`. Maybe

```
    return entityTables.entrySet().stream()
        .map(entry -> new TableMetadata().name(entry.getKey()).hasData(entry.getValue().getCount() != 0))
        .toList();
```
Also there should be a unit test for this case.